cmplibrary/src/main/java/com/sourcepoint/cmplibrary/data/local/DataStorageUSNat.kt

```diff
@@ -0,0 +1,49 @@
+package com.sourcepoint.cmplibrary.data.local
+
+import android.content.Context
+import android.content.SharedPreferences
+import android.preference.PreferenceManager
+import com.sourcepoint.cmplibrary.data.local.DataStorageUSNat.Companion.USNAT_CONSENT_STATUS
+
+internal interface DataStorageUSNat {
+
+    val preference: SharedPreferences
+
+    var usNatConsentStatus: String?
+
+    fun deleteUsNatConsent()
+
+    companion object {
+        const val USNAT_CONSENT_STATUS = "sp.usnat.key.consent.status"
+    }
+}
+
+internal fun DataStorageUSNat.Companion.create(
+    context: Context
+): DataStorageUSNat = DataStorageUSNatImpl(context)
+
+private class DataStorageUSNatImpl(context: Context) : DataStorageUSNat {
+
+    override val preference: SharedPreferences by lazy {
+        PreferenceManager.getDefaultSharedPreferences(context)
+    }
+
+    override var usNatConsentStatus: String?
```

Comment:
I think we should rename it to `usNatConsentData`, since the signature of the object is `UsNatConsentData`, and we should keep the naming convention straight, from my pov